docs/subsystems/api-release-checklist.md

```diff
@@ -1,43 +1,93 @@
 
-3. Create a [pure Python wheel][4].
+3. Follow PyPI's instructions in
+   [Generating distribution archives][generating-dist-archives] to generate the
+   archives for each package. It is recommended to manually inspect the build output
+   for the `zulip_bots` package to make sure that the extra files mentioned above
+   are not included in the archives.
 
-4. [Upload][5] the distribution file(s) to [zulip-beta][6].
+4. Follow PyPI's instructions in [Uploading distribution archives][upload-dist-archives]
+   to upload each package's archives to TestPyPI, which is a separate instance of the
+   package index intended for testing and experimentation. **Note**: You need to
+   [create a TestPyPI](https://test.pypi.org/account/register/) account for this step.
 
-5. Post about the beta release in `#general` and test
-   the [zulip-beta][6] package extensively.
+5. Follow PyPI's instructions in [Installing your newly uploaded package][install-pkg]
+   to test installing all three packages from TestPyPI.
 
-6. Respond to the feedback received in **Step 5**.
+6. If everything goes well in step 5, you may repeat step 4, except this time, upload
+   the packages to the actual Python Package Index.
 
-7. Make final changes, upload the distribution file(s) to the
-   main [zulip][1] package.
+7. Once the packages are uploaded successfully, set `IS_PYPA_PACKAGE` to `False` in
+   `zulip_bots/setup.py` and commit your changes with the version increments. Push
+   your commit to `python-zulip-api`. Create a release tag and push the tag as well.
+   See [the tag][example-tag] and [the commit][example-commit] for the 0.8.1 release
+   to see an example.
 
-8. Post in `#general` about the new release.
+Now it is time to update the dependencies in the [Zulip server repository][zulip-repo]:
 
-**Note:** We may upload directly to the main [zulip][1] package
-without beta-testing on [zulip-beta][6], if we feel that the changes
-made in the new release are minor and not disruptive enough to
-warrant extensive pretesting.
+1. Increment `PROVISION_VERSION` in `version.py`. A minor version bump should suffice in
+   most cases.
 
-[1]: https://pypi.python.org/pypi/zulip/0.3.1
-[2]: https://packaging.python.org/tutorials/distributing-packages/#configuring-your-project
-[3]: https://packaging.python.org/tutorials/distributing-packages/#source-distributions
-[4]: https://packaging.python.org/tutorials/distributing-packages/#pure-python-wheels
-[5]: https://packaging.python.org/tutorials/distributing-packages/#upload-your-distributions
-[6]: https://pypi.python.org/pypi/zulip-beta/0.2.5
+2. Update the release tags in the Git URLs for `zulip` and `zulip_bots` in
+   `requirements/common.in`.
+
+3. Run `tools/update-locked-requirements` to update the rest of the requirements files.
+
+4. Commit your changes and submit a PR and @-mention Tim Abbott so that he can merge
+   the PR! **Note**: See [this example commit][example-zulip-commit] to get an idea
```

Comment:
I don't think we need the mention in the checklist here; new PRs for dependency upgrades will be reviewed prompty regardless of whether they mention me.